src/main/java/br/com/autonomiccs/apacheCloudStack/client/ApacheCloudStackClient.java

```diff
@@ -189,7 +196,11 @@ protected HttpContext createHttpContextWithAuthenticatedSessionUsingUserCredenti
      *  If {@link #validateServerHttpsCertificate} indicates that we should not validate HTTPS server certificate, we use an insecure SSL factory; the insecure factory is created using {@link #createInsecureSslFactory()}.
      */
     protected CloseableHttpClient createHttpClient() {
-        HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
+        RequestConfig config = RequestConfig.custom()
```

Comment:
Instead of "1000", what about using the constant DateUtils.MILLIS_PER_SECOND?
Also, I think it is a good idea to extract lines 199-202, this enables docs (for instance explaining these three different timeout configurations that the method creates) and a unit test case.
Also, do not forget to add the verification on “br.com.autonomiccs.apacheCloudStack.client.ApacheCloudStackClientTest.configureExecuteAndVerifyTestForCreateHttpClient(boolean, int)” for a single method call for the newly created method that creates “RequestConfig” with connection timeouts.